tests/completions.sh

```diff
@@ -0,0 +1,48 @@
+source common.sh
+
+cd "$TEST_ROOT"
+
+mkdir -p dep && pushd dep
+cat <<EOF > flake.nix
+{
+    outputs = i: { };
+}
+EOF
+popd
+mkdir -p foo && pushd foo
+cat <<EOF > flake.nix
+{
+    inputs.a.url = "path:$(realpath ../dep)";
+
+    outputs = i: {
+        sampleOutput = 1;
+    };
+}
+EOF
+
+popd
+
+# Test the completion of a subcommand
+[[ $(printf "normal\nbuild\t\n") == $(NIX_GET_COMPLETIONS=1 nix buil) ]]
```

Comment:
```suggestion
[[ $(NIX_GET_COMPLETIONS=1 nix buil) == $'normal\nbuild\t\n' ]]
```

`==` isn't symmetrical, in that the LHS is treated as a single word so that you don't need to quote expansions, but the RHS is treated as a pattern so that unquoted glob characters will mess things up. Also, no need for a command substitution to get a fixed string.